lib/discourse_theme/cli_commands/rspec.rb

```diff
@@ -0,0 +1,242 @@
+# frozen_string_literal: true
+
+require "selenium-webdriver"
+
+module DiscourseTheme
+  module CliCommands
+    class Rspec
+      DISCOURSE_TEST_DOCKER_CONTAINER_NAME_PREFIX = "discourse_theme_test"
+      DISCOURSE_THEME_TEST_TMP_DIR = "/tmp/.discourse_theme_test"
+      SELENIUM_HEADLESS_ENV = "SELENIUM_HEADLESS=0"
+
+      class << self
+        def discourse_test_docker_container_name
+          "#{DISCOURSE_TEST_DOCKER_CONTAINER_NAME_PREFIX}_#{DiscourseTheme::VERSION}"
+        end
+
+        def run(settings:, dir:, args:, reset: false)
+          settings.local_discourse_directory = nil if reset
+
+          spec_path = "/spec"
+          index = dir.index(spec_path)
+
+          if index
+            spec_path = dir[index..-1]
+            dir = dir[0..index - 1]
+          end
+
+          spec_directory = File.join(dir, "/spec")
+
+          unless Dir.exist?(spec_directory)
+            raise DiscourseTheme::ThemeError.new "'#{spec_directory} does not exist"
+          end
+
+          configure_local_directory(settings)
+
+          headless = !!args.delete("--headless")
+
+          if settings.local_discourse_directory.empty?
+            run_tests_with_docker(
+              Pathname.new(dir).basename.to_s,
+              spec_directory,
+              spec_path,
+              headless: headless,
+              verbose: !!args.delete("--verbose"),
+              rebuild: !!args.delete("--rebuild"),
+            )
+          else
+            run_tests_locally(
+              settings.local_discourse_directory,
+              spec_directory,
+              headless: headless,
+            )
+          end
+        end
+
+        private
+
+        def execute(command:, message: nil, exit_on_error: true, stream: false)
+          UI.progress(message) if message
+
+          success = false
+          output = +""
+
+          Open3.popen2e(command) do |stdin, stdout_and_stderr, wait_thr|
+            Thread.new do
+              stdout_and_stderr.each do |line|
+                puts line if stream
+                output << line
+              end
+            end
+
+            exit_status = wait_thr.value
+            success = exit_status.success?
+
+            unless success
+              UI.error "Error occured while running: `#{command}`:\n\n#{output}" unless stream
+              exit 1 if exit_on_error
+            end
+          end
+
+          output
+        end
+
+        def run_tests_locally(local_directory, spec_path, headless: false)
+          UI.progress(
+            "Running RSpec tests using local Discourse repository located at '#{local_directory}'...",
+          )
+
+          Kernel.exec(
+            ENV,
+            "cd #{local_directory} && #{headless ? SELENIUM_HEADLESS_ENV : ""} bundle exec rspec #{spec_path}",
+          )
```

Comment:
I am wondering if it is better to rely more on Ruby and less on shell scripting:

```suggestion
          ENV["SELENIUM_HEADLESS"] = "0"
          Dir.chdir(local_directory)
          Kernel.exec("bundle exec rspec #{spec_path}")
```

Reply:
I tried this but using `Dir.chdir(local_directory)` ended up changing the working directory in the test environment. Will keep this as it is for now :)